src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/configs/CacheConfig.java

```diff
@@ -22,21 +22,41 @@
 
     private Double fieldDataCacheSizeThreshold;
     private Double shardRequestCacheSizeThreshold;
+    private Integer fieldDataCollectorTimePeriodInSec;
+    private Integer shardRequestCollectorTimePeriodInSec;
 
     public static final double DEFAULT_FIELD_DATA_CACHE_SIZE_THRESHOLD = 0.8;
     public static final double DEFAULT_SHARD_REQUEST_CACHE_SIZE_THRESHOLD = 0.9;
+    public static final int DEFAULT_FIELD_DATA_COLLECTOR_TIME_PERIOD_IN_SEC = 300;
+    public static final int DEFAULT_SHARD_REQUEST_COLLECTOR_TIME_PERIOD_IN_SEC = 300;
 
     public CacheConfig(final RcaConf rcaConf) {
         fieldDataCacheSizeThreshold = rcaConf.readRcaConfig(CONFIG_NAME,
                 RCA_CONF_KEY_CONSTANTS.FIELD_DATA_CACHE_SIZE_THRESHOLD, Double.class);
         shardRequestCacheSizeThreshold = rcaConf.readRcaConfig(CONFIG_NAME,
                 RCA_CONF_KEY_CONSTANTS.SHARD_REQUEST_CACHE_SIZE_THRESHOLD, Double.class);
+        fieldDataCollectorTimePeriodInSec =
```

Comment:
We should have some validation logic in these configs (beyond type checks), like not allowing negative values. I added this support with the `Config` object in PR #402 , it takes a `validator` predicate function. Pls consider leveraging it if relevant.